numpy/_core/src/multiarray/getset.c

```diff
@@ -124,6 +124,11 @@ array_strides_set(PyArrayObject *self, PyObject *obj, void *NPY_UNUSED(ignored))
     npy_intp upper_offset = 0;
     Py_buffer view;
 
+    /* DEPRECATED 2025-05-04, NumPy 2.3 */
+    PyErr_WarnEx(PyExc_DeprecationWarning,
+                "Setting the strides on a Numpy array has been deprecated in Numpy 2.3.\n",
```

Comment:
We could point to `strided_window_view` and `stride_tricks.as_strided`.  Although, not sure it is needed for this one.